### PR TITLE
Clarify security requirements for blinded claim names

### PR DESCRIPTION
**Pull request by danielfett**
Addresses Issue #132

### CODE REVIEW

**Comment by danielfett** [comment]
> Addresses Issue #132

I also revisited the security recommendations for the salt and made some smaller changes there.

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 1]
```diff
@@ -914,12 +922,18 @@ salt MUST be created in such a manner that it is cryptographically random,
...
-## Minimum length of the salt
+## Minimum length of the salt {#salt-minlength}
 
-The RECOMMENDED length of the randomly-generated portion of the salt is at least 128 bits.
+The RECOMMENDED minimum length of the randomly-generated portion of the salt is 128 bits.
 
 Note that minimum 128 bits would be necessary when SHA-256, HMAC-SHA256, or a function of similar strength is used, but a smaller salt size might achieve similar level of security if a stronger iterative derivation function is used.
 
+The issuer MUST ensure that a new salt value is chosen for each claim,
+including for claims that carry the same name at different places in the
```
Comment (Sakurann) [inline draft-ietf-oauth-selective-disclosure-jwt.md:932]:
```suggestion
including when the same claim name occurs at different places in the
```

File: draft-ietf-oauth-selective-disclosure-jwt.md [hunk 2]
```diff
@@ -933,15 +947,26 @@ revealed fundamental weaknesses and they MUST NOT be used.
...
 
+With the approach chosen in this specification, claim names of objects
+that are not themselves selectively disclosable are not blinded.  This
+can be seen in Example 6 in the Appendix, where even in the blinded
+SD-JWT, `address` and `delivery_address` are visible. This limitation
+needs to be taken into account by issuers when creating the structure of
+the SD-JWT.
+
+The issuer MUST ensure that a new random placeholder name is chosen for
+each claim, including for claims that carry the same name at different
```
Comment (Sakurann) [inline draft-ietf-oauth-selective-disclosure-jwt.md:963]:
```suggestion
each claim, including when the same claim name occurs at different
```

**Review by Sakurann — APPROVED** [review]
I agree with these changes, but it really made me think that we need to clarify various options we have introduced how claims in SD-JWT can be represented as it is becoming a bit hard to follow..

issue #159 filed
